syntax = "proto3";

package chat;

message ChatMessage {
  string user = 1;        // 发送者用户名
  int32 room_id = 2;      // 房间号
  string content = 3;     // 文本内容
  int64 timestamp = 4;    // 发送时间，毫秒
  MessageType type = 5;   // 消息类型
}

enum MessageType {
  UNKNOWN = 0;
  SYSTEM = 1;   // 系统提示（加入/离开）
  USER_TEXT = 2; // 用户文本
  QUERY_COUNT = 3; // 查询房间人数
  ROOM_COUNT = 4;  // 房间人数响应
  MUSIC = 5;    // 音乐消息
}
